src/gui/qgsexpressionbuilderwidget.cpp

```diff
@@ -440,6 +443,26 @@ void QgsExpressionBuilderWidget::loadRecent( const QString &collection )
   }
 }
 
+void QgsExpressionBuilderWidget::loadLayers()
+{
+  QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
```

Comment:
Yes, that's what's scaring me as well when I think that we ever would want to get rid of `QgsProject::instance`.